plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergFileOperations.java

```diff
@@ -197,6 +198,7 @@ public void testSelect()
                         .add(new FileOperation(SNAPSHOT, "InputFile.length"))
                         .add(new FileOperation(SNAPSHOT, "InputFile.newStream"))
                         .add(new FileOperation(MANIFEST, "InputFile.newStream"))
+                        .add(new FileOperation(MANIFEST, "InputFile.length"))
```

Comment:
I think @rdblue can add a comment on this. 

I haven't explored this change in depth. @nastra has helped in fixing these test failures. 
Let me understand it aswell.

Reply:
I think I see what's causing this change:
https://github.com/apache/iceberg/pull/9592/files#diff-2e0ffbcd66a9c894e677ff49fb760ccfb28191c7b7fa4e2dbe5f9b16d6a27560R184
we used to use the manifest.length() (the Iceberg maintained length), now we're using the[ object storage length ](https://github.com/apache/iceberg/pull/9592/files#diff-aeb2747061941360740730b65414fb2547bf560ce11762e02803e89dd76155a2R144)in tryCache. 

Not sure yet if that change is required for both encrypted and non-encrypted or just encrypted files.

Reply:
Stepped through the debugger before/after the upgrade, it is related to that specific change I linked but doesn't really have anything to do with the tryCache part (that's only relevant for manifest caching but we don't exercise that path in these tests). 

Prior to the change we would do an inputFile.length() but we'd just used the already passed in manifest length from Iceberg metadata, and Trino's custom `TracingInputFile` used in these tests would correctly not count that as a length call since it's cached, and not hitting the file system. 

Now that we're not passing in the manifest length, we are actually making the object storage requests to derive the length and the Trino `TracingInputFile` will correctly log that request as a filesystem access.

Reply:
Actually the problem is more that the default implementation of `FileIO#newInputFile(ManifestFile)` does not pass the length. 

We should fix the default implementation in Iceberg but we can avoid the regression in Trino by overriding `ForwardingFileIO#newInputFile(ManifestFile)` to have a correct implementation which passes the manifest length. @ajantha-bhat i'll raise a PR to your branch.

Reply:
Thanks @amogh-jahagirdar. I have merged it. 

Do we need to open a discussion on Iceberg mailing list to properly fix it. Those who don't override will still go through an extra IO? Manually overriding for each fileIO is not efficient. DO we have to revert/update the Iceberg side change and have 1.5.1?

Reply:
ping @danielcweeks, @rdblue

Reply:
I've opened a PR on the Iceberg side to address this at the default implementation level https://github.com/apache/iceberg/pull/9953. We just need the fix in the default implementation, not a revert of the original commit.

Feel free to raise your thoughts on the dev list, but my thinking is that this regression isn't particularly significant to warrant a patch release but I can understand the concern, and it definitely doesn't feel great to have that in. 

I think it's not significant because the extra IO is a head request done specifically when reading manifests (the overriden cases for `DataFile` and `DeleteFile` are not exercised, but I put them there to be defensive). HEAD requests are (in the most extreme cases) upper 2 digit milliseconds, typically though 10-20ms. Reading manifests is typically a very small amount of time compared to the overall job/query time for most workloads, and typically only a small amount of manifests are read to begin with. 

All this leads me to believe this probably isn't worth addressing in a patch release, but I can certainly be convinced otherwise. Perhaps I'm generalizing too much about folk's workloads and how different query engines will behave!

Reply:
Good find!

Can we get https://github.com/apache/iceberg/pull/9953 in?
Also, i think we should have static analysis for `ForwardingFileIo` -- https://github.com/trinodb/trino/pull/21079